File: Barath/src/04_Remove-Duplicates/removeDuplicates.ts
function removeDuplicates(nums: (number | string)[]): (number | string)[] {
    if (nums.length === 0) return nums;

    let index = 1;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) {
            nums[index] = nums[i];
            index++;
        }
    }

    for (let i = index; i < nums.length; i++) {
        nums[i] = '_';
    }

    return nums;
}

console.log(removeDuplicates([1,1,2]));



export default removeDuplicates;

File: Bipin/020_remove_duplicates/remove_duplicates.go
package main

// RemoveDuplicates removes duplicates in-place from a sorted array and returns the new length.
func RemoveDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	// Initialize two pointers: `i` for the unique elements
	i := 0

	// Iterate through the array starting from the second element
	for j := 1; j < len(nums); j++ {
		// If the current element is different from the previous unique element
		if nums[j] != nums[i] {
			i++
			nums[i] = nums[j]
		}
	}

	// Return the length of the array with unique elements
	return i + 1
}
